File: BOJ(Java 공부용)/1181.java
import java.io.*;
import java.util.ArrayList;

public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine().split(" ")[0]);
        ArrayList<p> v = new ArrayList<>();
        for(int i=0;i<n;i++) {
            String s = br.readLine();
            v.add(new p(s,s.length()));
        }
        v.sort((a,b)->{
           if(a.length<b.length) return -1;
           else if(a.length>b.length) return 1;
           else {
               return a.s.compareTo(b.s);
           }
        });
        String temp = v.get(0).s;
        bw.write(temp+"\n");
        for(int i=1;i<v.size();i++) {
            if(temp .equals(v.get(i).s)) continue;
            bw.write(v.get(i).s+"\n");
            temp = v.get(i).s;
        }
        bw.flush();
        bw.close();
    }
    private static class p{
        String s;
        int length;
        private p(String s,int length) {
            this.s = s;
            this.length = length;
        }
    }
}

File: BOJ(Kotlin)/1181.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*
import kotlin.math.max
import kotlin.math.min

var ans = 0
fun main(args: Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    val st = StringTokenizer(br.readLine())
    val n = st.nextToken().toInt()
    var list : MutableList<WordList> = mutableListOf()
    for (i in 1..n) {
        val item = br.readLine()
        list.add(WordList(item,i))
    }
   list.sortWith { o1, o2 ->
       when {
           o1.alpha.length < o2.alpha.length -> -1
           o1.alpha.length == o2.alpha.length -> {
               o1.alpha.compareTo(o2.alpha)
           }
           else -> 1
       }
   }
    var setList = mutableSetOf<String>()
    for (value in list) {
        setList.add(value.alpha)
    }
    for (value in setList) {
        bw.write(value + "\n")
    }
    bw.flush()
    bw.close()
}

class WordList(val alpha : String, val index : Int) {}